code/modules/atmospherics/gasmixtures/gas_mixture.dm

```diff
@@ -431,27 +431,33 @@ GLOBAL_LIST_INIT(gaslist_cache, init_gaslist_cache())
 			var/datum/gas_reaction/reaction = r
 
 			var/list/min_reqs = reaction.min_requirements.Copy()
-			if((min_reqs["TEMP"] && temp < min_reqs["TEMP"]) \
-			|| (min_reqs["ENER"] && ener < min_reqs["ENER"]))
+			var/min_temp = min_reqs["TEMP"]
+			var/min_ener = min_reqs["ENER"]
+			if((!isnull(min_temp) && temp < min_temp) \
+			|| (!isnull(min_ener) && ener < min_ener))
 				continue
-			min_reqs -= "TEMP"
-			min_reqs -= "ENER"
-
+			var/static/list/temp_and_energy = list("TEMP", "ENER")
+			min_reqs -= temp_and_energy

```

Comment:
I'm gonna profile this one myself and if it makes a difference I'll pr it. Not the others tho